app/helpers/application_helper.rb

```diff
@@ -92,11 +92,13 @@ def link_to_add_fields(name, f, association)
   # label.  For anything more precise, a custom helper should be
   # used (as in items_helper.rb)
   def success_or_warning_label(str, success_str)
```

Comment:
it returns html for:

```<span class="label label-success">str</span>```
or
```<span class="label label-warning">str</span>```
depending on whether str matches success_str, so it does actually add a label.